### PR TITLE
CI: fix docker pushes

### LINKED ISSUE #41: Missing Docker version tags
The [latest commit](https://github.com/wemake-services/caddy-gen/commit/5f9437af8af56a558cde76ea3dd052f63b24e29b) includes breaking changes. Which caused a server of mine to become inaccessible. Since there's no image tagged with version 0.3.0 on Docker Hub I can't easily revert.

Also a tag, possible 0.4.0, for the aforementioned commit/release would be nice.

### PR DESCRIPTION
**Pull request by casperdcl**
Fixes #41

ref | image
--|--
PR | - (just test)
`master` | `wemakeservices/caddy-gen:master`, `ghcr.io/wemake-services/caddy-gen:master`
tag `M.m.p` | `wemakeservices/caddy-gen:M.m.p`, `wemakeservices/caddy-gen:latest`, `ghcr.io/wemake-services/caddy-gen:M.m.p`, `ghcr.io/wemake-services/caddy-gen:latest`

### CODE REVIEW

**Comment by casperdcl** [comment]
@sobolevn this is ready for merge now. It'll push to `ghcr.io/wemake-services/caddy-gen`. If you want to push to `wemakeservices/caddy-gen` too, then you'll have to add `DOCKER_USER` and `DOCKER_PASS` env vars to https://github.com/wemake-services/caddy-gen/settings/secrets/actions

**Review by sobolevn — COMMENTED** [review]
You can copy release workflow from https://github.com/wemake-services/kira-review

**Comment by casperdcl** [comment]
done ;)

**Comment by casperdcl** [comment]
ping :)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,30 +1,66 @@
...
-      - master
+    branches: [master]
+    tags: ['**']
   pull_request:
   workflow_dispatch:
-
-concurrency: 
-  group: ${{ github.head_ref || github.run_id }}
+concurrency:
+  group: ${{ github.ref }}
```
Comment (sobolevn) [inline .github/workflows/test.yml:9]:
```suggestion
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,30 +1,66 @@
...
     steps:
     - uses: actions/checkout@v4
-    - name: Set up Python
-      uses: actions/setup-python@v5
       with:
-        python-version: '3.9'
-    
-    - name: Run tests
+        fetch-depth: 0
+        ref: ${{ github.event.pull_request.head.sha || github.ref }} # fix SHA
```
Comment (sobolevn) [inline .github/workflows/test.yml:21]:
```suggestion
```

**Review by sobolevn — COMMENTED** [review]
Sorry, I missed the notification

File: .github/workflows/test.yml (hunk 1 above)
Comment (casperdcl) [reply .github/workflows/test.yml:9]: I don't quite understand this suggestion.

Currently whenever a branch/tag/PR/ is updated, any previous jobs on the same branch/tag/PR are cancelled.

Do you want something else? Note that[^conc][^gh]:

- there are no other workflows so `github.workflow` isn't needed
- `github.head_ref` is only defined on PRs
- `github.run_id` would have no effect (it's the default behaviour)

[^conc]: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#concurrency
[^gh]: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts#github-context

**Comment by sobolevn** [comment]
Thank you!

**Pull request merged**